src/backer/cardaning.py

```diff
@@ -104,122 +98,130 @@ def flushQueue(self):
             self.timer = Timer(90, self.flushQueue)
             self.timer.start()
 
-    def getaddressBalance(self):
-        try:
-            address = self.api.address(address=self.spending_addr.encode())
-            return int(address.amount[0].quantity)
-        except ApiError as e:
-            return 0
+    def getaddressBalance(self, addr):
+        try:            
+            utxo_list = self.client.query_utxo.execute([ogmios.Address(addr)])       
+
+            if utxo_list and len(utxo_list[0]) > 0:          
+                utxo = utxo_list[0][0]                
+                return int(utxo.value['ada']['lovelace'])
+            
+        except Exception as e:
+            logger.debug(traceback.format_exc())
+            logger.info(f"Error: Cannot get address's balance: {e}")
+        
+        return 0
 
     def fundAddress(self, addr):
         try:
-            funding_payment_signing_key = PaymentSigningKey.from_cbor(os.environ.get('FUNDING_ADDRESS_CBORHEX'))
-            funding_payment_verification_key = PaymentVerificationKey.from_signing_key(funding_payment_signing_key)
-            funding_addr = Address(funding_payment_verification_key.hash(), None, network=NETWORK)
+            funding_payment_signing_key = pycardano.PaymentSigningKey.from_cbor(os.environ.get('FUNDING_ADDRESS_CBORHEX'))
+            funding_payment_verification_key = pycardano.PaymentVerificationKey.from_signing_key(funding_payment_signing_key)
+            funding_addr = pycardano.Address(funding_payment_verification_key.hash(), None, network=NETWORK)
         except KeyError:
             print("Backer address could not be funded. Environment variable FUNDING_ADDRESS_CBORHEX is not set")
             return
+        
         try:
-            fund_addr = funding_addr.encode()
-            resp = self.api.address(address=fund_addr)
-            funding_balance = resp.amount[0]
-        except ApiError as e:
-            if e.status_code == 404:
-                print("Empty balance to fund backer")
-            else:
-                print("Blockfrost API error:",e.message)
+            fund_addr = funding_addr.encode()            
+            funding_balance = self.getaddressBalance(fund_addr)
+        except Exception as e:
+            logger.info("Empty balance to fund backer")
```

Comment:
I think we can remove the try catch here. It's handled in `getaddressBalance`

Reply:
done